client/lib/abtest/index.js

```diff
@@ -234,21 +234,15 @@ ABTest.prototype.getSavedVariation = function() {
 };
 
 ABTest.prototype.assignVariation = function() {
-	let variationName, randomAllocationAmount, userid;
+	let variationName, randomAllocationAmount;
 	let sum = 0;
 
 	const allocationsTotal = reduce( this.variationDetails, ( allocations, allocation ) => {
 		return allocations + allocation;
 	}, 0 );
 
-	if ( typeof user !== 'undefined' && typeof user.data !== 'undefined' ) {
-		userid = Number( user.data.ID );
-	} else {
-		userid = Number( 'noUserObject' );
-	}
-
-	if ( this.assignmentMethod === 'userid' && ! isNaN( userid ) ) {
-		randomAllocationAmount = userid % allocationsTotal;
+	if ( this.assignmentMethod === 'userid' && typeof user !== 'undefined' && typeof user.data !== 'undefined' && ! isNaN( Number( user.data.ID ) ) ) {
```

Comment:
Rather than `Number( user.data.ID )`, you can prefix with `+` to coerce to a number:

<img width="135" alt="screen shot 2017-07-12 at 19 38 01" src="https://user-images.githubusercontent.com/17325/28134026-a29b4be2-6739-11e7-86fc-3ee7ae5f8bf6.png">

You could grab the user ID with lodash `get` the line before:

`const userId = get( user, 'data.ID' );`

so you can do something like...

`if ( this.assignmentMethod === 'userid' && ! isNaN( +userId ) ) ) {`